pyschism/forcing/hycom/hycom2schism.py

```diff
@@ -727,104 +763,207 @@ def fetch_data(self, outdir: Union[str, os.PathLike], hgrid, vgrid, start_date,
 
 class DownloadHycom:
 
-    def __init__(self, hgrid):
-
-        xmin, xmax = hgrid.coords[:, 0].min(), hgrid.coords[:, 0].max()
-        ymin, ymax = hgrid.coords[:, 1].min(), hgrid.coords[:, 1].max()
+    def __init__(self, hgrid=None):
+        if hgrid is not None:
+            xmin, xmax = hgrid.coords[:, 0].min(), hgrid.coords[:, 0].max()
```

Comment:
In practice it would be much better to use a `bbox` argument rather than `hgrid` argument, because it is not practical to be forced to load an `hgrid` in order to just download some hycom data. On the hand hand, the user can now instantiate without an `hgrid` and change the `bbox` property on the instance, which I guess also works.